src/core/Akka.Cluster/Cluster.cs

```diff
@@ -78,6 +79,9 @@ public Cluster(ActorSystemImpl system)
 
             _scheduler = CreateScheduler(system);
 
+            // it has to be lazy - otherwise if downing provider will init a cluster itself, it will deadlock
+            _downingProvider = new Lazy<IDowningProvider>(() => Akka.Cluster.DowningProvider.Load(Settings.DowningProviderType, system), LazyThreadSafetyMode.ExecutionAndPublication);
```

Comment:
As we don't have control over how `IDowningProvider` is actually implemented, I wouldn't set `PublicationOnly` - from the programmer point of view it would be safe to assume, that his/her provider won't be created random number of times, especially as it might be backed by potentially heavy initialization logic.